### DISCUSSION TITLE
Generate profile from instance/model

### DISCUSSION THREAD

**Discussion by TheTimKiely** [cimug-org/CIMTool#61] [Q&A]
Hi,
I'm very new to using CIM, so please correct me if I'm misunderstanding something.
I have a CIM model exported from OSII Monarch.  In order to use the SISCO CIM Adapter to import this model into OSISoft PI AF, I think I need to create a profile representing all the classes and properties used in that model.  
It seems unsustainable to create the profile manually in CIMTool.
Is there any way to automate the generation of a profile from a model?  I could parse the model and generate a "data definition" spreadsheet.  Then create a profile from that data definition.  Would that be the best way to go about automated profile generation?
Thanks,
Tim

**Comment by tviegut** [comment]
> Hi,
> 
> I'm very new to using CIM, so please correct me if I'm misunderstanding something.
> 
> I have a CIM model exported from OSII Monarch.  In order to use the SISCO CIM Adapter to import this model into OSISoft PI AF, I think I need to create a profile representing all the classes and properties used in that model.  
> 
> It seems unsustainable to create the profile manually in CIMTool.
> 
> Is there any way to automate the generation of a profile from a model?  I could parse the model and generate a "data definition" spreadsheet.  Then create a profile from that data definition.  Would that be the best way to go about automated profile generation?
> 
> Thanks,
> 
> Tim

I apologize as I'm traveling so am responding late here. Aspects of your question are not entirely clear to me.  You're question is framed within the context of an export from OSII Monarch. I assume that to be a CIM/XML instance file containing a static network model. I do not see how you would want to use instance data to create a profile in CIMTool (?). Stepping back, a CIMTool profile is derived off of a CIM schema (e.g. XMI file) that you import into CIMTool. If you want to create a "profile" you would really want to parse the XMI. I apologize if my response here is not helpful. Perhaps if you could provide greater detail. What is not clear is what the import and export file formats are that you need across the various tools you reference (SISCO, OSISoft).
